Use AssetManagerAndQuarantinedFileStorage for attachments

chrisroos opened the pull request:
This PR configures the `AttachmentUploader` to store attachments on the file system and in Asset Manager. This addresses https://github.com/alphagov/asset-manager/issues/440.

I've tested this change as working in the development vm. I created a consultation and uploaded a single attachment and a single image. The attachment was created in the draft state while the images were created in a non-draft state:

We're not going to be ready to serve these attachment assets from Asset Manager for a while so marking them as draft seems like the safest option in the meantime.

floehopper commented on app/workers/asset_manager_create_whitehall_asset_worker.rb:
```diff
@@ -1,7 +1,9 @@
   def perform(file_path, legacy_url_path)
     file = File.open(file_path)
-    Services.asset_manager.create_whitehall_asset(file: file, legacy_url_path: legacy_url_path)
+    asset_options = {file: file, legacy_url_path: legacy_url_path}
```
Very minor: It's usual to have spaces after the opening brace and before the closing brace of a Hash. I would've expected Rubocop to complain about this.

chrisroos replied:
Rubocop did complain! I fixed it but put it in the subsequent commit by mistake. I've now updated this commit.

floehopper commented on app/workers/asset_manager_create_whitehall_asset_worker.rb:
```diff
@@ -1,7 +1,9 @@
     file = File.open(file_path)
-    Services.asset_manager.create_whitehall_asset(file: file, legacy_url_path: legacy_url_path)
+    asset_options = {file: file, legacy_url_path: legacy_url_path}
+    asset_options[:draft] = true if legacy_url_path =~ /attachment_data/
```
Minor: I'm not keen on trailing `if` statements - I think the code is clearer if you use the block form.

chrisroos replied:
I generally agree if it's a long line but I'm happy that this is short enough not to be too unreadable so I'm going to leave it as is.

floehopper commented on app/workers/asset_manager_create_whitehall_asset_worker.rb:
```diff
@@ -1,8 +1,8 @@
-    asset_options = {file: file, legacy_url_path: legacy_url_path}
-    asset_options[:draft] = true if legacy_url_path =~ /attachment_data/
+    asset_options = { file: file, legacy_url_path: legacy_url_path }
+    asset_options[:draft] = true if legacy_url_path.match?(/attachment_data/)
```
Minor: These changes seem orthogonal to the commit note. I think they'd be better combined into the first commit.

chrisroos replied:
Good spot. I incorrectly added it to this commit. I've now moved it to the previous commit.

floehopper commented on test/unit/whitehall_uploader_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class WhitehallUploaderTest < ActiveSupport::TestCase
+  test 'indicates that assets are protected' do
```
Typo: I think this should be "indicates that assets are *not* protected"

chrisroos replied:
Correct! I've updated the test name.

floehopper commented:
@chrisroos: One other thought. It would be worth explaining somewhere what the rationale is for marking *all* new attachments as draft, irrespective of their state in Whitehall. Presumably we'll need to update the draft status of these assets when we've implemented https://github.com/alphagov/asset-manager/issues/447.

chrisroos commented:
> One other minor thought: it would be nice to use keyword arguments for worker perform method now there are three of them. However, I imagine the switchover might cause problems with queued jobs, so perhaps it's not worth it.

I agree that keyword args would be better but that switching over would be a bit painful so I'm not going to do anything about this for now.

chrisroos commented:
> One other thought. It would be worth explaining somewhere what the rationale is for marking all new attachments as draft, irrespective of their state in Whitehall.

I've added a note to the commit that marks these attachments as draft in asset-manager. I've added the same note to the description of this PR.

> Presumably we'll need to update the draft status of these assets when we've implemented alphagov/asset-manager#447.

Yes. Or at least before we finally switch the nginx config to serve these attachments.

chrisroos commented:
Thanks for the review, @floehopper. I've addressed your comments, rebased on master and force pushed in preparation for merging.
